language/solana/move-to-solana/src/stackless/dwarf.rs

```diff
@@ -598,6 +806,19 @@ impl<'up> DIBuilder<'up> {
 
         if let Some(_di_builder_core) = &self.0 {
             let di_builder = self.builder_ref().unwrap();
+            let module_di_test = &self.module_di().unwrap();
+            let module_di = &m_ctx.llvm_module.0;
+            assert!(
+                module_di == module_di_test,
+                "Module context should be the same"
+            );
+            unsafe {
+                LLVMDumpModule(*module_di);
+            }
+            let module_di_info = print_module_to_str(module_di);
+            debug!(target: "dwarf", "\ncreate_vector bof DI starting at next line and until line starting with !!!\n{module_di_info}\n!!!\n");
```

Comment:
This line goes into the debug output, **!!!** helps to find the end of printed debug_info message, since it is a multiline. 
Here are a few examples:

```Code
[DEBUG language/solana/move-to-solana/src/stackless/dwarf.rs:500] DIBuilder bof DI starting at next line and until line starting with !!!
; ModuleID = '0x101__vector'
source_filename = "0x101__vector"

!!!
```
```Code
[DEBUG language/solana/move-to-solana/src/stackless/dwarf.rs:597] DIBuilder bof DI starting at next line and until line starting with !!!
; ModuleID = '0x101__vector'
source_filename = "/home/sol/tmp/vector.2.move"

!llvm.dbg.cu = !{!0}

!0 = distinct !DICompileUnit(language: DW_LANG_Rust, file: !1, producer: "move-mv-llvm-compiler", isOptimized: false, runtimeVersion: 0, emissionKind: FullDebug, splitDebugInlining: false, sysroot: "/")
!1 = !DIFile(filename: "vector.2.move", directory: "/home/sol/tmp")

!!!
```

Reply:
DI easily accepts any garbage in parameters. So when garbage is printed it helps to see where it comes from and for this I use this EOM marker.
Please notice that we are discussing something what printed only in the debug messages, it is not coming to the actual debug_info.